lib/explorer/polars_backend/shared.ex

```diff
@@ -9,33 +9,36 @@ defmodule Explorer.PolarsBackend.Shared do
   alias Explorer.PolarsBackend.Series, as: PolarsSeries
   alias Explorer.Series, as: Series
 
-  def apply_native(df_or_s, fun, args \\ [])
+  defguardp is_polars_df(df) when is_struct(df, PolarsDataFrame) or is_struct(df, PolarsLazyFrame)
+  defguardp is_polars_series(series) when is_struct(series, PolarsSeries)
```

Comment:
Nitpick: the pattern matching is much more efficient for checking structs... so I would remove the guards. So even though there is duplication, I think the duplication may be worth it.

Reply:
If you want to encapsulate it, then maybe `@polars_df [PolarsDataFrame, PolarsLazyFrame]` is better. :)

Reply:
Good to know!

Using `%module{} when module in @polars_df` is equivalent to having a function clause for each, right?
https://github.com/elixir-nx/explorer/pull/237/commits/a09ee9d43b93318d17cc8464684c394e9e53450d

Reply:
In terms of performance it would be pretty close, yeah!